### PR TITLE
chore: Udate OTel and prometheus components

### PR DESCRIPTION
**Pull request by ridwanmsharif**
This change does the following:

- Updates OTel components
- Removes the untyped metrics hack, and replace it with the support
  upstream (using metric metadata instead of an additional attribute per
  sample)
- Confgenerator updates
- Several test updates

This change brings our fork much closer to upstream OTel and Prometheus.
This should be pretty helpful when unforking.

### CODE REVIEW

**Comment by ridwanmsharif** [comment]
Related library change: https://github.com/GoogleCloudPlatform/prometheus/tree/feature/shutdownscrape_v0.54.1

File: collector/receiver/prometheusreceiver/internal/transaction.go [hunk 1]
```diff
@@ -191,6 +188,11 @@ func (t *transaction) AppendHistogram(ref storage.SeriesRef, l labels.Labels, at
 	return 0, nil
 }
 
+func (t *transaction) AppendCTZeroSample(_ storage.SeriesRef, _ labels.Labels, _, _ int64) (storage.SeriesRef, error) {
+	//TODO: implement this func
```
Comment (braydonk) [inline collector/receiver/prometheusreceiver/internal/transaction.go:192]: We're okay without this I presume?

File: confgenerator/testdata/add-metadata-labels/golden/otel.yaml [hunk 2]
```diff
@@ -100,37 +104,46 @@ receivers:
...
           target_label: cluster
           replacement: __run__
           action: replace
         - source_labels: [__address__]
+          regex: null
           target_label: namespace
           replacement: test_service
           action: replace
         - source_labels: [__address__]
+          regex: null
```
Comment (braydonk) [inline confgenerator/testdata/add-metadata-labels/golden/otel.yaml:140]: The `regex: null` shouldn't affect anything right?

**Review by braydonk — APPROVED** [review]

File: collector/receiver/prometheusreceiver/internal/transaction.go (hunk 1 above)
Comment (ridwanmsharif) [reply collector/receiver/prometheusreceiver/internal/transaction.go:192]: Yup this is something we are working on on the CT working group - its a way of adding explicit resets to the Prometheus TSDB. This doesn't affect the collector at all

**Review by jefferbrecht — APPROVED** [review]

**Pull request merged**